### PR TITLE
TFModel improvements

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
This PR proposes to:

- Make `ConvBlock` able to chain multiple layers, just like `Torch` version can

- Simplify logic of letter parsing, as well as adding capability of using `R` letter as separate `Branch` with complex parameters

- Swap all arguments inside calls to `conv_block` to keyword ones

- Add `squeeze-and-excitation` versions of `ResNet`

- Re-check all the tests of model compilation

- Add various attention modules: some of them are available through `S` (stands for self-attention) letter, some of them are mods of `Combine`

### CODE REVIEW

File: batchflow/models/tf/layers/core.py [hunk 1]
```diff
@@ -66,32 +81,83 @@ class Combine(Layer):
 
     data_format : str {'channels_last', 'channels_first'}
         Data format.
+    leading_index : int
+        Index of tensor to broadcast to.
```
Comment (a-arefina) [inline batchflow/models/tf/layers/core.py]: Please add description

File: batchflow/models/tf/layers/conv.py [hunk 2]
```diff
@@ -107,6 +108,7 @@ def __call__(self, inputs):
 @add_as_function
 class ConvTranspose(Layer):
     """ Transposed Nd convolution layer.
+    Used for `C` letter in layout convention of :class:`~.tf.layers.ConvBlock`.
```
[the remark this answers is not in the document]
Comment (SergeyTsimfer) [reply batchflow/models/tf/layers/conv.py]: It is indeed `t` letter

File: batchflow/models/tf/base.py [hunk 3]
```diff
@@ -969,9 +969,11 @@ def _fill_feed_dict(self, feed_dict=None, device=None, is_training=True):
         _feed_dict = {}
         for placeholder, value in feed_dict.items():
             if self.has_classes(placeholder):
-                classes = self.classes(placeholder)
-                get_indices = np.vectorize(lambda c, arr=classes: np.where(c == arr)[0])
-                value = get_indices(value)
+                classes = self.get_tensor_config(placeholder).get('classes')
+                if not isinstance(classes, int):
```
[the remark this answers is not in the document]
Comment (SergeyTsimfer) [reply batchflow/models/tf/base.py]: Currently, `TFModel` is capable of working with custom class labels: [`cats`, `dogs`] for example. In this case, we need to manually transform such labels into int-classes, and this transform takes a lot of time. We don't need to do this in the case of `integer` classes.

**Comment by SergeyTsimfer** [comment]
[make ConvBlock great again](https://trello.com/c/qki67yia/13-make-convblock-great-again)

**Comment by SergeyTsimfer** [comment]
[Tutorial for advanced TF features](https://trello.com/c/JlDFQvMS/17-tutorial-for-advanced-tf-features)
